include/ddc/kernels/splines/splines_linear_problem_pds_band.hpp

```diff
@@ -0,0 +1,172 @@
+ * @brief A positive-definite symmetric band linear problem dedicated to the computation of a spline approximation.
+ *
+ * The storage format is dense row-major. Lapack is used to perform every matrix and linear solver-related operations.
+ *
+ * Given the linear system A*x=b, we assume that A is a square (n by n)
+ * with kd sub and superdiagonals.
+ * All non-zero elements of A are stored in the rectangular matrix q, using
+ * the format required by DPBTRF (LAPACK): (super-)diagonals of A are rows of q.
+ * q has 1 row for the diagonal and kd rows for the superdiagonals.
+ *
+ * @tparam ExecSpace The Kokkos::ExecutionSpace on which operations related to the matrix are supposed to be performed.
+ */
+template <class ExecSpace>
+class SplinesLinearProblemPDSBand : public SplinesLinearProblem<ExecSpace>
+{
+public:
+    using typename SplinesLinearProblem<ExecSpace>::MultiRHS;
+    using SplinesLinearProblem<ExecSpace>::size;
+
+protected:
+    Kokkos::View<double**, Kokkos::HostSpace> m_q; // pds band matrix representation
+
+public:
+    /**
+     * @brief SplinesLinearProblemPDSBand constructor.
+     *
+     * @param mat_size The size of one of the dimensions of the square matrix.
+     * @param kd The number of sub/superdiagonals of the matrix.
+     */
+    explicit SplinesLinearProblemPDSBand(std::size_t const mat_size, std::size_t const kd)
+        : SplinesLinearProblem<ExecSpace>(mat_size)
+        , m_q("q", kd + 1, mat_size)
+    {
+        assert(m_q.extent(0) <= mat_size);
+
+        Kokkos::deep_copy(m_q, 0.);
+    }
+
+    virtual double get_element(std::size_t const i, std::size_t const j) const override
+    {
+        assert(i < size());
+        assert(j < size());
+
+        if (i == j) {
+            return m_q(0, i);
+        }
+
+        // Indices are swapped for an element on subdiagonal
+        std::size_t i_tmp;
+        std::size_t j_tmp;
+        if (i < j) {
+            i_tmp = i;
+            j_tmp = j;
+        } else {
+            i_tmp = j;
+            j_tmp = i;
+        }
+        for (int k = 1; k < m_q.extent(0); ++k) {
+            if (i_tmp + k == j_tmp) {
+                return m_q(k, i_tmp);
+            }
+        }
+        return 0.0;
+    }
+
+    virtual void set_element(std::size_t const i, std::size_t const j, double const aij) override
+    {
+        assert(i < size());
+        assert(j < size());
+
+        if (i == j) {
+            m_q(0, i) = aij;
+            return;
+        }
+
+        // Indices are swapped for an element on subdiagonal
+        std::size_t i_tmp;
+        std::size_t j_tmp;
+        if (i < j) {
+            i_tmp = i;
+            j_tmp = j;
+        } else {
+            i_tmp = j;
+            j_tmp = i;
+        }
+        for (int k = 1; k < m_q.extent(0); ++k) {
+            if (i_tmp + k == j_tmp) {
+                m_q(k, i_tmp) = aij;
+                return;
+            }
+        }
+        assert(std::fabs(aij) < 1e-20);
+        return;
+    }
```

Comment:
Would something like this work ? Same question for `get_element`

```suggestion
    virtual void set_element(std::size_t i, std::size_t j, double const aij) override
    {
        assert(i < size());
        assert(j < size());

        // Indices are swapped for an element on subdiagonal
        if (i > j) {
            std::swap(i, j);
        }

        if (j - i < m_q.extent(0)) {
            m_q(j - i, i) = aij;
        } else {
            assert(std::fabs(aij) < 1e-20);
        }
    }
```

Reply:
Why do you need a loop ?